data/binding/preference.go

```diff
@@ -119,3 +195,79 @@ func (b *prefBoundBase[T]) checkForChange() {
 func (b *prefBoundBase[T]) replaceProvider(p fyne.Preferences) {
 	b.get, b.set = b.setLookup(p)
 }
+
+type prefBoundList[T bool | float64 | int | string] struct {
+	boundList[T]
+	key string
+
+	get       func(string) []T
+	set       func(string, []T)
+	setLookup preferenceLookupSetter[[]T]
+}
+
+func (b *prefBoundList[T]) checkForChange() {
+	val := *b.val
+	updated := b.get(b.key)
+	if val == nil || len(updated) != len(val) {
+		b.Set(updated)
+		return
+	}
+	if val == nil {
+		return
+	}
+
+	// incoming changes to a preference list are not at the child level
+	for i, v := range val {
+		if i >= len(updated) {
+			break
+		}
+
+		if !b.comparator(v, updated[i]) {
+			_ = b.items[i].(Item[T]).Set(updated[i])
+		}
+	}
+}
+
+func (b *prefBoundList[T]) replaceProvider(p fyne.Preferences) {
+	b.get, b.set = b.setLookup(p)
+}
+
+type internalPrefs = interface{ WriteValues(func(map[string]any)) }
+
+func bindPreferenceListComparable[T bool | float64 | int | string](key string, p fyne.Preferences,
```

Comment:
If we are to be consistent with the other functions here, the one called "comparable" should just pass a hard-coded `func(T, T) bool` into the function without the "comparable" suffix. Doing the same here would avoid the code duplication of all the comparison functions you have above.

Reply:
Ah good catch thanks.